synapse/models/infotech.py

```diff
@@ -71,9 +65,211 @@ def getBaseModels():
+                    ('exe', {'ptype': 'file:bytes', 'doc': 'The file containing the code that read the file'}),
+
+                )),
+
+                ('it:exec:file:write', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('path', {'ptype': 'file:path'}),
+                    ('path:dir', {'ptype': 'file:path', 'ro': 1}),
+                    ('path:ext', {'ptype': 'str:lwr', 'ro': 1}),
+                    ('path:base', {'ptype': 'file:base', 'ro': 1}),
+                    ('file', {'ptype': 'file:bytes'}),
+                    ('time', {'ptype': 'time'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('exe', {'ptype': 'file:bytes', 'doc': 'The file containing the code that wrote to the file'}),
+
+                )),
+
+                ('it:exec:reg:get', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
+
+                ('it:exec:reg:set', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
+
+                ('it:exec:reg:del', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
             ),
 
         }
-        name = 'it'
-        return ((name, modl), )
+
+        models = (
+            ('it', modl),
+        )
+
+        return models
+
+    def intiCoreModule(self):
+        pass
+
+    @on('node:form', form='it:dev:str')
```

Comment:
we have the onFormNode helper in CoreModule for this instead of using an @on helper.

Reply:
i've been debating it's removal...  we can circle up about this, but lets not block the PR